chain/cosmos/poll.go

```diff
@@ -58,3 +59,24 @@ func PollForMessage[T any](ctx context.Context, chain *CosmosChain, registry cod
 	bp := testutil.BlockPoller[T]{CurrentHeight: chain.Height, PollFunc: doPoll}
 	return bp.DoPoll(ctx, startHeight, maxHeight)
 }
+
+// PollForBalance polls until the balance matches
+func PollForBalance(ctx context.Context, chain *CosmosChain, deltaBlocks uint64, balance ibc.WalletAmount) error {
```

Comment:
This is really helpful!! I'm all for moving to a poll based approach throughout `ibctest` vs. waiting arbitrary block heights and hoping the timing works out.